### PR TITLE
Implement .toHaveValue() with tests and docs

### PR DESCRIPTION
**Pull request by dreewp**
Related Issue: #22 
- Implements .toHaveValue() Jest matcher method
- Adds tests to handle multiple cases
- Adds a README that includes the multiple cases
- Ran npm test to make sure nothing breaks
- Ran `node docs/build` to add to `index.html`

Wasn't sure if matching multiple elements was necessary, but seems like a method that would be targeted for a single element, let me know if you think otherwise!

### CODE REVIEW

File: src/jest-matchers/toHaveValue/test.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  it('requires only one element', () => {
+    const $list = $(
+      <li>
+        <input type="text" value="first" readOnly />
+        <input type="text" value="second" readOnly />
+      </li>
+    );
+    expect(() => expect($list.find('input')).toHaveValue('first')).toThrow(
+      'Cannot check multiple elements for values. Please pass only one element.'
+    );
```
Comment (franciscop) [inline src/jest-matchers/toHaveValue/test.js:65]: Nice 👍

**Comment by franciscop** [comment]
> Wasn't sure if matching multiple elements was necessary, but seems like a method that would be targeted for a single element, let me know if you think otherwise!

Totally agree with that assessment, and it's nice to see that it gives a clear error.

**Comment by franciscop** [comment]
As a general recommendation for documentation, it should normally start light and then go deeper as you keep reading. Also, start from the common usecase and then go deeper into the gotchas and edge cases. If the docs start with a block of code, it might be overwhelming for newer users trying to get started with the library.

So for example here it could have started with an intro like:

```md
Checks whether the form element has the given value:

```js
const $input = $(<input type="text" value="this is an example" onChange={} />);
expect($input).toHaveValue("this is an example");
\```

It only works on the input, textarea, and select:

```js
// Example for textarea and select
// Possibly the textarea with `defaultValue` instead of `value`, or maybe that should be an example later on
\```

Except for inputs of type checkbox and radio, for those please use `.checked` instead:

```js
// Failing example for <input type="checkbox" />
\```
```

**Comment by dreewp** [comment]
@franciscop I added a defaultValue test and some docs for now, but may need editing later on.

**Pull request merged**